seagoat/file.py

```diff
@@ -20,6 +21,20 @@ def __repr__(self):
     def add_commit(self, commit_hash: str):
         self.commit_hashes.add(commit_hash)
 
+    def get_encoding(self) -> str | None:
+        detector = UniversalDetector()
+        file_bin = open(self.absolute_path, "rb")
+
+        for line in file_bin:
+            detector.feed(line)
+            if detector.done: break
+        detector.close()
+        if detector.result['confidence'] < 0.4:
+            return 'bin'
```

Comment:
hmm, so if the encoding cannot be detected, we assume that it is binary? I can see that this is a safe option in terms of not breaking the server, however if we have logic to make the server "resilient to errors" I would prefer to not do it here but in some generic place 

```suggestion
            return 'utf-8'
```

Reply:
If you are going to handle encoding errors in some other way, then better to always return `detector.result['encoding']`, as it is likelier to be the correct encoding than utf-8.

Reply:
you are right, that makes sense to me! and then, it would also simplify the code as checking for the confidence would not be necessary anymore, right?